src/libraries/Common/src/System/Net/ArrayBuffer.cs

```diff
@@ -54,7 +54,7 @@ public void Dispose()
         }
 
         public int ActiveLength => _availableStart - _activeStart;
-        public Span<byte> ActiveSpan => new Span<byte>(_bytes, _activeStart, _availableStart - _activeStart);
+        public ReadOnlySpan<byte> ActiveSpan => new Span<byte>(_bytes, _activeStart, _availableStart - _activeStart);
```

Comment:
Please keep as mutable; I take advantage of this in HTTP/3.

Reply:
Sounds like this should remain Span for Cory's needs.  If it turns out you do change the type, you should also change the type being constructed; no point in causing more work for the JIT to need to undo the construction and then implicit conversion from Span to ReadOnlySpan.

Reply:
Should then I use Span inside of PAL, or should I add ReadOnblySPan to Buffers, or should I use ReadOnlySpan consistently when I can and pay penalty for conversion in places where it does not work?

Reply:
Updated. I added ActiveReadOnlySpan as that is cheap and easy and gives us the most flexibility.